components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticator.java

```diff
@@ -172,11 +173,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             }
             idpSubject = (String) request.getSession().getAttribute("username");
             if (subject == null) {
+                if (idpSubject == null) {
+                    throw new SAMLSSOException("Cannot find federated User Identifier");
+                }
```

Comment:
Well, we cannot do this. In this case we won't be able to configure the UserIdClaim Uri in the identity provider configuration and use that as the subject.